packages/datadog-plugin-mocha/test/index.spec.js

```diff
@@ -374,32 +374,21 @@ Timeout of 100ms exceeded. For async tests and hooks, ensure "done()" is called;
       it('works with retries', (done) => {
         const testFilePath = path.join(__dirname, 'mocha-test-retries.js')
 
-        let numTestSpans = 0
```

Comment:
retries have changed the way they work. I have it aligns better with how `mocha` works: however many times a test is retried, it's still considered a single test span. If it ends up passing, no information about the error is sent, just like `mocha` works: it reports nothing if a retried test ends up passing. Making it work liked it used to required a lot of hacks and I think this is just better and cleaner